### PR TITLE
reconfigured apache to listen on port 8080 - so we no longer need roo…

### PR DESCRIPTION
**Pull request by devfaz**
…t privileged to start

Fixes issue https://github.com/firefly-iii/firefly-iii/issues/3449

Changes in this pull request:
- use port 8080 instead of 80

@JC5

### CODE REVIEW

**Comment by JC5** [comment]
Thanks, I appreciate the effort. It works as expected. I'll pick up the merge next weekend probably, I have a first version (with your code copy-pasted). If it all looks to be working as expected, I can merge it.

Problem is, I can't go back. Switching to port 8080 is *huge*. Everybody using Firefly III needs to update everything and it's not just the two of us. If I'm going in this direction it has to be final, it has to work, *and no other solution needs to present itself*.

I still want to know what's causing root to write to the cache dir.

**Comment by devfaz** [comment]
Well, we could introduce some Port Variable. If its unset (we use 80 and root), if its set to >1024 we run without root. Further we will add the variable with value 8080 to the compose file and mark the setting as dreprecated in 6.x release.

**Comment by JC5** [comment]
Something optional is really out of the question. That only adds confusion and I fear a lot of questions. 

This change could be set up for the 5.3.x range. It’s a move to PHP 7.4 anyway so this won’t look out of place.

**Comment by JC5** [comment]
Merged!

File: Dockerfile.amd64 [hunk 1]
```diff
@@ -20,10 +20,14 @@ RUN a2enmod rewrite && a2enmod ssl && \
...
     sed -i 's/upload_max_filesize = 2M/upload_max_filesize = 64M/' /usr/local/etc/php/php.ini && \
-    sed -i 's/post_max_size = 8M/post_max_size = 64M/' /usr/local/etc/php/php.ini
+    sed -i 's/post_max_size = 8M/post_max_size = 64M/' /usr/local/etc/php/php.ini && \
+    chgrp www-data /var/log/apache2 && \
+    ln -svf /dev/stderr /var/log/apache2/error.log && \
+    ln -svf /dev/stdout /var/log/apache2/access.log
 
 COPY [ "./cacert.pem", "/usr/local/ssl/cert.pem" ]
 COPY [ "./apache2.conf", "/etc/apache2/apache2.conf" ]
+COPY [ "./apache2-ports.conf", "/etc/apache2/ports.conf" ]
```
Comment (SuperSandro2000) [inline Dockerfile.amd64:30]: This could simply be replaced with ``echo "Listen 8080">/etc/apache2/ports.conf``.

File: Dockerfile.amd64 [hunk 2]
```diff
@@ -20,10 +20,14 @@ RUN a2enmod rewrite && a2enmod ssl && \
     sed -i 's/max_execution_time = 30/max_execution_time = 600/' /usr/local/etc/php/php.ini && \
     sed -i 's/memory_limit = 128M/memory_limit = 512M/' /usr/local/etc/php/php.ini && \
     sed -i 's/upload_max_filesize = 2M/upload_max_filesize = 64M/' /usr/local/etc/php/php.ini && \
-    sed -i 's/post_max_size = 8M/post_max_size = 64M/' /usr/local/etc/php/php.ini
+    sed -i 's/post_max_size = 8M/post_max_size = 64M/' /usr/local/etc/php/php.ini && \
+    chgrp www-data /var/log/apache2 && \
+    ln -svf /dev/stderr /var/log/apache2/error.log && \
+    ln -svf /dev/stdout /var/log/apache2/access.log
```
Comment (SuperSandro2000) [inline Dockerfile.amd64:26]: You forgot to change that for the other arches.

File: apache-firefly-iii.conf [hunk 3]
```diff
@@ -1,4 +1,4 @@
-<VirtualHost *:80>
+<VirtualHost *:8080>
```
Comment (SuperSandro2000) [inline apache-firefly-iii.conf:1]: Thanks for probably breaking all setups.

**Pull request merged**